src/test/java/com/puppycrawl/tools/checkstyle/gui/ParseTreeTablePModelTest.java

```diff
@@ -0,0 +1,119 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2016 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.puppycrawl.tools.checkstyle.gui;
+
+import java.io.File;
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import com.puppycrawl.tools.checkstyle.TreeWalker;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.api.DetailNode;
+import com.puppycrawl.tools.checkstyle.api.FileContents;
+import com.puppycrawl.tools.checkstyle.api.FileText;
+import com.puppycrawl.tools.checkstyle.api.JavadocTokenTypes;
+import com.puppycrawl.tools.checkstyle.api.TokenTypes;
+
+import com.puppycrawl.tools.checkstyle.gui.MainFrameModel.ParseMode;
+
+public class ParseTreeTablePModelTest {
```

Comment:
please update pom.xml and remove gui classes from exclude by cobertura. Your coverage should be enforced by build process.

https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L1866
we need to remove that exclusion for all files in gui folder and use it only for swing related classes. All other classes should be covered by 100% or partly(https://github.com/checkstyle/checkstyle/blob/master/pom.xml#L1851) .

As we start writing UTs, nobody is allowed to reduce coverage - should be enforced by build process :) .